### PR TITLE
[CLI] Update/Remove/Rename CLI flags

### PR DESCRIPTION
**Pull request by IfSentient**
Clean up CLI flags by renaming some and removing others. Some functionality is also changed.

## Flag Changes
* `--selectors|-s` has been removed, and partially replaced by the `--manifest` flag. `--manifest` allows you to specify a selector for the app manifest, but does not allow for multiple selectors like `--selectors` did. This change was made as we no longer source kinds, but a manifest, and there should be one manifest per app.
* `--cuepath|-c` has been renamed to `--source|-s`, to better reflect the future goal of having different source formats vi the `--format` flag.
* `--crdencoding` has been renamed to `--defencoding`, as it now encodes both CRDs and app manifests
* `--crdpath` has been renamed to `--defpath`, as it dictates the location of more than just CRDs now (also app manifest)
* `--kindgrouping` has been renamed to `--grouping` to simplify (though I'm not 100% sure this is a positive change, perhaps just a neutral one to make the flag name shorter)

## Generated Code Changes

The `/resource` component of the generated code when grouping by kinds has been removed. To preserve this behavior in an existing project where you don't want the path to change, use `-g=pkg/generated/resource` (instead of the default `pkg/generated`).

Drive-by fix to generated operator and backend plugin code imports when grouping by group.

Local dev environment now also respects the `--manifest` flag (it didn't respect `--selectors` before).

This PR also contains a drive-by fix to `app.MultiRunner` for writing to a closed channel.

### CODE REVIEW

File: app/runner.go [hunk 1]
```diff
@@ -56,7 +56,11 @@ func (m *MultiRunner) Run(ctx context.Context) error {
 			err := r.Run(propagatedContext)
 			wg.Done()
 			if err != nil && !timedOut {
-				errs <- err
+				select {
```
Comment (IfSentient) [inline app/runner.go:59]: THis is a drive-by fix for an edge case that crops up every so often in tests where the channel is already closed because the runner has exited, but the goroutine hadn't finished yet.

File: cmd/grafana-app-sdk/generate.go [hunk 2]
```diff
@@ -233,7 +203,7 @@ func generateKindsCue(modFS fs.FS, cfg kindGenConfig, selectors ...string) (code
 	// Manifest
 	var manifestFiles codejen.Files
 	var goManifestFiles codejen.Files
-	if cfg.GenerateManifest {
+	if cfg.CRDEncoding != "none" {
 		if cfg.CRDEncoding != "none" {
```
Comment (radiohead) [inline cmd/grafana-app-sdk/generate.go:207]: Coulde we simplify this?

File: cmd/grafana-app-sdk/project.go [hunk 3]
```diff
@@ -411,53 +397,53 @@ func projectAddComponent(cmd *cobra.Command, args []string) error {
...
 	}
 
+	manifests, err := manifestParser.Parse(os.DirFS(sourcePath), selector)
+	if err != nil {
+		return fmt.Errorf("error parsing manifest '%s': %v", sourcePath, err)
+	}
+	if len(manifests) == 0 {
+		return fmt.Errorf("no manifest found in '%s'", sourcePath)
+	}
+	manifest := manifests[0]
```
Comment (radiohead) [inline cmd/grafana-app-sdk/project.go:438]: Any chance we get `len(manifests) > 1`?

File: cmd/grafana-app-sdk/project.go [hunk 4]
```diff
@@ -555,7 +541,7 @@ func addComponentOperator[G anyGenerator](projectRootPath string, generator G, s
 func addComponentBackend[G anyGenerator](projectRootPath string, generator G, selectors []string, pluginID string, groupKinds bool) error {
 	// Check plugin ID
 	if pluginID == "" {
-		return fmt.Errorf("plugin-id is required")
+		return fmt.Errorf("manifest group")
```
Comment (radiohead) [inline cmd/grafana-app-sdk/project.go:544]:
```suggestion
		return fmt.Errorf("manifest group is required")
```

File: cmd/grafana-app-sdk/project.go [hunk 5]
```diff
@@ -555,7 +541,7 @@ func addComponentOperator[G anyGenerator](projectRootPath string, generator G, s
 func addComponentBackend[G anyGenerator](projectRootPath string, generator G, selectors []string, pluginID string, groupKinds bool) error {
```
Comment (radiohead) [inline cmd/grafana-app-sdk/project.go:541]: NIT, but not required:
```suggestion
func addComponentBackend[G anyGenerator](projectRootPath string, generator G, selectors []string, manifestGroup string, groupKinds bool) error {
```

File: cmd/grafana-app-sdk/project.go (hunk 3 above)
Comment (IfSentient) [reply cmd/grafana-app-sdk/project.go:438]: There's not a normal situation where this should happen (we only allow one selector, and the parser should only return one), if we do, taking the first one should be fine, unless we'd rather error.

Comment (radiohead) [reply cmd/grafana-app-sdk/project.go:438]: I think we could error (for future-proofing, since explicit failure is easier to debug than weird behaviour) but I don't think it's a blocker – feel free to leave it as-is.

**Comment by IfSentient** [comment]
I'll merge this in after today's release
